Src/IronPython/Modules/_bytesio.cs

```diff
@@ -146,59 +145,39 @@ public BytesIO(CodeContext/*!*/ context, object initial_bytes=null)
             [Documentation("readinto(array_or_bytearray) -> int.  Read up to len(b) bytes into b.\n\n"
                 + "Returns number of bytes read (0 for EOF)."
                 )]
-            public BigInteger readinto([NotNull]ByteArray buffer) {
-                _checkClosed();
-
-                int len = Math.Min(_length - _pos, buffer.Count);
-                for (int i = 0; i < len; i++) {
-                    buffer[i] = _data[_pos++];
-                }
-
-                return len;
-            }
-
-            public BigInteger readinto([NotNull]ArrayModule.array buffer) {
-                _checkClosed();
+            public BigInteger readinto([NotNull] IBufferProtocol buffer) {
+                try {
+                    using var pythonBuffer = buffer.GetBuffer(BufferFlags.Writable);
```

Comment:
Any ideas how to make this try/catch nicer? I was thinking a `TryGetBuffer` extension method might do the trick.

Reply:
I agree, it would make the code nicer. Though it will "try" though a try/catch, so no performance improvement.

Reply:
Not sure if it's any better:
```C#
if (!buffer.TryGetBuffer(out IPythonBuffer pythonBuffer, flags: BufferFlags.Writable)) {
    throw PythonOps.TypeError("readinto() argument must be read-write bytes-like object, not {0}", PythonTypeOps.GetName(buffer));
}
using (pythonBuffer) {
    // ...
}
```

Reply:
Wow, this does look weird to me. Maybe I am just not used to such a pattern.

Reply:
Yeah, not quite sure how else I'd use a `using` with a try pattern. Maybe this instead:
```C#
using var pythonBuffer = buffer.GetBufferNoThrow(BufferFlags.Writable)
    ?? throw PythonOps.TypeError("readinto() argument must be read-write bytes-like object, not {0}", PythonTypeOps.GetName(buffer));
```

Reply:
To me, this feels much better.